frontend/src/components/user/ATSUserCreateModal.vue

```diff
@@ -0,0 +1,159 @@
+<script setup lang="ts">
+import axios from 'axios';
+import { onMounted, ref } from 'vue';
+
+import { Spinner } from '@/components/layout';
+import { Button, Column, DataTable, Dialog, useToast } from '@/lib/primevue';
+import { atsService } from '@/services';
+import { BasicResponse, Initiative } from '@/utils/enums/application';
+import { setEmptyStringsToNull } from '@/utils/utils';
+
+import type { Ref } from 'vue';
+import type { ATSUser, ATSUserResponse, Submission } from '@/types';
+
+// Props
+const { submission } = defineProps<{
+  submission: Submission;
+}>();
+
+// Emits
+const emit = defineEmits(['atsUserLink:link']);
+
+// State
+const atsClientNumber: Ref<string> = ref('');
+const loading: Ref<boolean> = ref(false);
+const proponent: Ref<ATSUser | undefined> = ref(undefined);
+const selectedUser: Ref<ATSUser | undefined> = ref(undefined);
+const visible = defineModel<boolean>('visible');
+
+// Actions
+
+const toast = useToast();
+
+async function createATSClient() {
+  selectedUser.value = undefined;
+  try {
+    loading.value = true;
+    const data = {
+      '@type': 'ClientResource',
+      address: {
+        '@type': 'AddressResource',
+        addressLine1: submission.streetAddress,
+        city: submission.locality,
+        provinceCode: submission.province,
+        primaryPhone: submission.contactPhoneNumber,
+        email: submission.contactEmail
+      },
+      firstName: submission.contactFirstName,
+      surName: submission.contactLastName,
+      regionName: Initiative.HOUSING,
+      optOutOfBCStatSurveyInd: BasicResponse.NO.toUpperCase()
+    };
+
+    const submitData: ATSUserResponse = setEmptyStringsToNull(data);
```

Comment:
This is not acting as a response type. Can `ATSUser` be used here?

Reply:
type changed